From 6002e02ead769e116f2a0dfbe2e4a2517d26aa85 Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Wed, 12 Jun 2024 10:03:51 +0100
Subject: [PATCH 30/41] video_chroma/chain: Fix chroma+resize then transform

* Would be worth finding a test case for this


Upstream-Status: Inappropriate

RPI-Distro repo forks original vlc and applies patches

to enable raspiberry pi support.
---

 modules/video_chroma/chain.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/modules/video_chroma/chain.c
+++ b/modules/video_chroma/chain.c
@@ -280,8 +280,9 @@ static int BuildTransformChain( filter_t
         return VLC_SUCCESS;
 
     /* Lets try resize+chroma first, then transform */
-    msg_Dbg( p_filter, "Trying to build chroma+resize" );
-    EsFormatMergeSize( &fmt_mid, &p_filter->fmt_out, &p_filter->fmt_in );
+    msg_Dbg( p_filter, "Trying to build chroma+resize, then transform" );
+    es_format_Copy( &fmt_mid, &p_filter->fmt_out );
+    video_format_TransformTo(&fmt_mid.video, p_filter->fmt_in.video.orientation);
     i_ret = CreateChain( p_filter, &fmt_mid );
     es_format_Clean( &fmt_mid );
     if( i_ret == VLC_SUCCESS )
